libhb/hb_json.c

```diff
@@ -538,7 +539,13 @@ hb_dict_t* hb_job_to_dict( const hb_job_t * job )
     // necessary PAR value
 
     subtitle_search_burn = job->select_subtitle_config.dest == RENDERSUB;
-
+#if HB_PROJECT_FEATURE_QSV
+#if defined(_WIN32) || defined(__MINGW32__)
+    adapter_index = hb_qsv_get_adapter_index();
```

Comment:
I don't understand why the adapter_index is reset when converting from job to dict and back. Shouldn't it stay the same? Or if it needs to be set before starting the job, it would be a better place to do it in work.c or something similar.

Reply:
@galad87 thanks for your comments.

There are two stages:

1. Discovery stage where we collect information about all QSV adapters in the system. During this stage we select default QSV adapter with highest generation. This step is required because we actually don't know which adapter user select to run the job. There is no job for GUI app at that stage.
2. Run job stage where we actually read and apply the user selected adapter number from the job in hb_dict_to_job. No reset adapter when doing hb_job_to_dict.

I hope that explains my idea.

Reply:
@galad87 adapter logic moved to work.c as you suggested.